vnext/Microsoft.ReactNative.Cxx/JSValue.h

```diff
@@ -607,6 +615,10 @@ inline JSValue::JSValue(TBool value) noexcept : m_type{JSValueType::Boolean}, m_
 template <class TInt, std::enable_if_t<std::is_integral_v<TInt> && !std::is_same_v<TInt, bool>, int>>
 inline JSValue::JSValue(TInt value) noexcept : m_type{JSValueType::Int64}, m_int64{static_cast<int64_t>(value)} {}
 inline JSValue::JSValue(double value) noexcept : m_type{JSValueType::Double}, m_double{value} {}
+template <class T>
+inline JSValue::JSValue(std::optional<T> &&value) noexcept
+    : JSValue(value.has_value() ? JSValue(std::move(value).value()) : JSValue()) {}
```

Comment:
If I understand this correctly this is equating an empty `value`, with null; shouldn't it be equating it with `undefined` instead? are null and undefined the same as far as JSValue knows? and if so is that right?

Reply:
The JSValue is design to represent a JSON value with some design features from `folly:dynamic` such as having `Int64` and `Double` types instead of a single `Number` type. It only recognizes `Null`. It does not support the `undefined` type.